### PR TITLE
Fix deploying latency monitor

### PR DESCRIPTION
**Pull request by ausias-armesto**
- Adds push gateway url

## Summary by CodeRabbit

- **New Features**
	- Enhanced application observability by integrating with a Prometheus push gateway for latency monitoring.
	- Updated configuration settings to include new environment variables for improved metrics collection, including mandatory checks for critical variables.
	- Introduced a `.prettierignore` file to exclude sensitive files from automated formatting.
	- Added support for dynamic labeling of metrics based on environment variables.
	- Automated the merging of pull requests and Docker image builds through new GitHub Actions workflows.
	- Implemented a manual release process with version bumping and notifications.

- **Improvements**
	- Streamlined workflow configuration for the build process, including clearer steps for repository checkout, Node.js setup, and Google Cloud Platform integration.
	- Improved metrics collection process for better performance analysis and debugging.
	- Excluded Visual Studio Code settings from version control to maintain project cleanliness.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -11,6 +11,8 @@ type Settings = {
     pushGateway?: string;
     intervalMs: number;
     offsetMs: number;
+    metricLabels: { [key: string]: string };
```
Comment (esterlus) [inline src/index.ts]: would be more concise:

```
    metricLabels: Record<string, string>;
```

File: src/index.ts [hunk 2]
```diff
@@ -11,6 +11,8 @@ type Settings = {
...
+    metrics: { [key: string]: prom.Summary };
```
Comment (esterlus) [inline src/index.ts]: same here: 

```
    metrics: Record<string, prom.Summary>;
```

File: src/index.ts [hunk 3]
```diff
@@ -57,15 +59,86 @@ if (require.main === module) {
...
+        settings.metricLabels['location'] = process.env.UHTTP_LM_METRIC_LOCATION;
+    }
+
+    if (process.env.UHTTP_LM_METRIC_LATITUDE !== undefined) {
+        settings.metricLabels['latitude'] = process.env.UHTTP_LM_METRIC_LATITUDE;
+    }
+
+    if (process.env.UHTTP_LM_METRIC_LONGITUDE !== undefined) {
+        settings.metricLabels['longitude'] = process.env.UHTTP_LM_METRIC_LONGITUDE;
+    }
```
Comment (esterlus) [inline src/index.ts]: I would prefer to move those to line 55, so all the env var handling is in one place

File: src/index.ts [hunk 4]
```diff
@@ -85,63 +158,33 @@ function start(uHTTPsettings: UHTTPsettings, settings: Settings) {
...
+        metrics['rpcSum'].observe(metricLabels, metricsDurations.rpcDur);
+        metrics['exitAppSum'].observe(metricLabels, metricsDurations.exitAppDur);
+        metrics['segSum'].observe(metricLabels, metricsDurations.segDur);
+        metrics['hoprSum'].observe(metricLabels, metricsDurations.hoprDur);
     };
 }
 
-function reportError(hops: number) {
+function reportError(
+    metrics: { [key: string]: prom.Summary },
```
Comment (esterlus) [inline src/index.ts]: prefer `Record` syntax as commented earlier

File: src/index.ts [hunk 5]
```diff
@@ -85,63 +158,33 @@ function start(uHTTPsettings: UHTTPsettings, settings: Settings) {
...
+    metricLabels: { [key: string]: string },
```
Comment (esterlus) [inline src/index.ts]: prefer `Record` syntax as commented earlier

**Review by esterlus — COMMENTED** [review]
Tested it and it works great.

Unfortunately we cannot really print the payload pushed to the push gateway with a reasonable effort.

File: src/index.ts (hunk 1 above)
Comment (ausias-armesto) [reply src/index.ts]: Changed

File: src/index.ts (hunk 2 above)
Comment (ausias-armesto) [reply src/index.ts]: Changed

File: src/index.ts (hunk 5 above)
Comment (ausias-armesto) [reply src/index.ts]: Changed

File: src/index.ts (hunk 4 above)
Comment (ausias-armesto) [reply src/index.ts]: Changed

File: src/index.ts (hunk 3 above)
Comment (ausias-armesto) [reply src/index.ts]: Moved

**Review by esterlus — APPROVED** [review]

**Review by esterlus — APPROVED** [review]
Great

**Pull request merged**